packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx

```diff
@@ -520,6 +478,472 @@ export default function OSCALCatalogBaseline() {
+              onClick={loadPage}
+            >
+              CATALOGS & BASELINES
+            </Typography>
+          </OSCALTertiaryGrayscaleButton>
+          <CatalogTooltip title={usedTitle.toUpperCase()}>
+            <Typography
+              color="primary"
+              sx={{
+                fontWeight: (theme) => theme.typography.fontWeightBold,
+                textTransform: "uppercase",
+                ":hover": {
+                  color: "#ff6600",
+                },
+                fontFamily: "Source Sans Pro",
+              }}
+            >
+              {trunckatedTitle.toUpperCase()}
+            </Typography>
+          </CatalogTooltip>
+        </OSCALBreadCrumbs>
+      );
+    else if (uploadNewCatalogBaseline) {
+      return (
+        <OSCALBreadCrumbs
+          aria-label="breadcrumb"
+          maxItems={3}
+          itemsBeforeCollapse={1}
+          itemsAfterCollapse={2}
+        >
+          <OSCALTertiaryGrayscaleButton>
+            <Typography
+              sx={{
+                ":hover": {
+                  color: "#ff6600",
+                },
+              }}
+              onClick={loadPage}
+            >
+              Home
+            </Typography>
+          </OSCALTertiaryGrayscaleButton>
+          <OSCALTertiaryGrayscaleButton>
+            <Typography
+              sx={{
+                ":hover": {
+                  color: "#ff6600",
+                },
+              }}
+              onClick={loadPage}
+            >
+              CATALOGS & BASELINES
+            </Typography>
+          </OSCALTertiaryGrayscaleButton>
+          <Typography
+            color="primary"
+            sx={{
+              fontWeight: (theme) => theme.typography.fontWeightBold,
+              textTransform: "uppercase",
+              ":hover": {
+                color: "#ff6600",
+              },
+              fontFamily: "Source Sans Pro",
+            }}
+          >
+            UPLOAD
+          </Typography>
+        </OSCALBreadCrumbs>
+      );
+    }
+    return (
+      <OSCALBreadCrumbs
+        aria-label="breadcrumb"
+        maxItems={3}
+        itemsBeforeCollapse={1}
+        itemsAfterCollapse={2}
+      >
+        {/* <OSCALLink href="/">HOME</OSCALLink> */}
+        <OSCALTertiaryGrayscaleButton>
+          <Typography
+            sx={{
+              ":hover": {
+                color: "#ff6600",
+              },
+            }}
+            onClick={loadPage}
+          >
+            Home
+          </Typography>
+        </OSCALTertiaryGrayscaleButton>
+        <Typography
+          color="primary"
+          sx={{
+            fontWeight: (theme) => theme.typography.fontWeightBold,
+            textTransform: "uppercase",
+            ":hover": {
+              color: "#ff6600",
+            },
+            fontFamily: "Source Sans Pro",
+          }}
+        >
+          CATALOGS & BASELINES
+        </Typography>
+      </OSCALBreadCrumbs>
+    );
```

Comment:
I'm trying to pick out the differences between these three conditions here, because it looks like a lot of reusable code. Just an idea, but can you just validate these conditions when attempting to render parts of this code? For example, with the first condition it looks like you're trying to figure out whether to include this breadcrumb item:
```
        {(item?.model.title?.length > 0) &&
          <CatalogTooltip title={usedTitle.toUpperCase()}>
            <Typography
              color="primary"
              sx={{
                fontWeight: (theme) => theme.typography.fontWeightBold,
                textTransform: "uppercase",
                ":hover": {
                  color: "#ff6600",
                },
                fontFamily: "Source Sans Pro",
              }}
            >
              {trunckatedTitle.toUpperCase()}
            </Typography>
          </CatalogTooltip>
         }

```